### PR TITLE
Play Spotify songs a user is listening to

### PR DESCRIPTION
**Pull request by MG-LSJ**
- Skip logic updated
- Added a command to play Spotify songs a user is listening to

### CODE REVIEW

File: examples/advance_music_cog.py [hunk 1]
```diff
@@ -469,6 +469,32 @@ async def previous(self, ctx, index: int = None):
         if previous_player := await self.MusicManager.previous(ctx, index):
             await ctx.send(f"Rewinding from {previous_player[0].title}")
 
+    # Spotify song from user
+    @commands.command()
+    async def play_user_spotify(self, ctx, user: discord.Member = None):
+        user = user if user else ctx.guild.get_member(ctx.author.id)
```
Comment (adam757521) [inline examples/advance_music_cog.py:475]: Please rename user to member for clarity.
Unnecessary to get the author id from the guild. Perhaps do this instead:
```py
member = member if member else ctx.author
```

File: examples/advance_music_cog.py [hunk 2]
```diff
@@ -469,6 +469,32 @@ async def previous(self, ctx, index: int = None):
...
+        spotify_result = next((activity for activity in user.activities if isinstance(activity, discord.Spotify)), None)
+
+        if spotify_result:
+            await ctx.send(f'{user.name} is not listening to Spotify.')
+            return
+        
+        query = f"{spotify_result.title} {spotify_result.artist}"
+
+        player = await self.MusicManager.create_player(query, user)
```
Comment (adam757521) [inline examples/advance_music_cog.py:484]: The requester should not be the user, but the ctx.author.

**Review by adam757521 — CHANGES_REQUESTED** [review]
This looks very good, the skip command has been fixed, but you do not have a play function/method, please add that and look at the comments I have left.

File: examples/advance_music_cog.py (hunk 2 above)
Comment (MG-LSJ) [reply examples/advance_music_cog.py]: fixed

**Comment by MG-LSJ** [comment]
Made a separate play function.

**Comment by MG-LSJ** [comment]
Added ctx.typing()

**Review by adam757521 — APPROVED** [review]

**Pull request merged**